src/Transactions/TransactionDetails/TransactionInformation/SourceValue/useSourceValue.js

```diff
@@ -0,0 +1,34 @@
+import { useQuery } from 'react-query';
+
+import { useOkapiKy } from '@folio/stripes/core';
+import {
+  LINES_API,
+  INVOICES_API,
+} from '@folio/stripes-acq-components';
+
+import { TRANSACTION_SOURCE } from '../../../constants';
+
+const queryFnsMap = {
+  [TRANSACTION_SOURCE.invoice]: async (ky, transaction) => {
+    const { folioInvoiceNo } = await ky(`${INVOICES_API}/${transaction.sourceInvoiceId}`).json();
+
+    return folioInvoiceNo;
+  },
+  [TRANSACTION_SOURCE.poLine]: async (ky, transaction) => {
+    const { poLineNumber } = await ky(`${LINES_API}/${transaction.encumbrance?.sourcePoLineId}`).json();
+
+    return poLineNumber;
+  },
+  [TRANSACTION_SOURCE.user]: () => TRANSACTION_SOURCE.user,
+  [TRANSACTION_SOURCE.fiscalYear]: () => TRANSACTION_SOURCE.fiscalYear,
+};
+
+export const useSourceValue = (transaction) => {
+  const ky = useOkapiKy();
+
+  const queryFn = queryFnsMap[transaction.source];
+
+  const result = useQuery(['finance', 'transaction-source-value', transaction.id], () => queryFn(ky, transaction));
```

Comment:
queryFn might be null

Reply:
might be. in that case `data: undefined`